src/dmd/semantic3.d

```diff
@@ -885,7 +885,6 @@ private extern(C++) final class Semantic3Visitor : Visitor
                 sc2.flags = (sc2.flags & ~SCOPEcontract) | SCOPErequire;
 
                 // BUG: need to error if accessing out parameters
-                // BUG: need to treat parameters as const
                 // BUG: need to disallow returns and throws
                 // BUG: verify that all in and ref parameters are read
```

Comment:
This comment as seen today appeared in D1.011.

Before it used to be:
```
BUG: verify that all in and inout parameters are read
```
Which has been here since the dawn of ~time~ svn history.

Reply:
It seems to allude to this:
```
int foo(in int a, ref int b)
in
{
    // Should emit error: 'a' is not read in contract.
    // Should emit error: 'b' is not read in contract.
}
do
{
    return a + b;
}
```
But I don't think that is a bug at all, and so this comment should be removed.  @WalterBright do I understand the context correctly?

Reply:
Why should an error be emitted if a parameter isn't read in the in-contract? How else would you express that any possible value is permitted for that parameter?

Reply:
Some languages enforce that some parameter are used, e.g. C# forces you to assign `out` parameters. One way to express this in an `in` contract would be to do `cast(void)a;` for example. But that's the kind of enforcement which IMO goes against D's approach, as many times you'll end up just trying to satisfy the compiler.

Reply:
I'm saying I don't think it should be considered as a bug.  Also the comments doesn't say anything about all parameters.  Just `in` and `ref` parameters.

Reply:
> verify that all in and ref parameters are read

It may also be implying that all input parameters should actually be used in the implementation.